### DISCUSSION TITLE
Linux Flatpak distribution

### DISCUSSION THREAD

**Discussion by digitalethics** [edrlab/thorium-reader#1750] [Ideas]
Could you make a Flatpak available via [Flathub](https://flathub.org)? I don't think AppImages are a sustainable solution for end-users for they are neither sandboxed nor offer any kind of convenient update mechanism.

**Comment by VaiTon** [comment]
Is this feature on the roadmap? It would mean a lot for linux users! Thanks!

Also, electron-builder does support Flatpak: https://www.electron.build/configuration/flatpak.html

**Comment by danielweck** [comment]
Relevant code: https://github.com/edrlab/thorium-reader/blob/2ad37fa8d4753c308b6d39b3f7c9439e09d93447/package.json#L217-L227

**Comment by danielweck** [comment]
Related: https://github.com/edrlab/thorium-reader/discussions/1740

**Reply by danielweck** [reply]
Related: https://github.com/edrlab/thorium-reader/discussions/1755

**Comment by danielweck** [comment]
From https://github.com/edrlab/thorium-reader/issues/1895

@Nitrousoxide

Flatpak is generally seeing a huge uptick as the app distribution method of choice for linux and it is distro agnostic. I know right now you have .deb releases, and of course you can always build it from source, but flatpak will let you control the dependencies you want installed and the baseline image your app is installed on so you can more easily ensure a standardized environment for it to run on while expanding the reach to more folks than just those running debian based OS's (or those who run it in a distrobox container)

Would it be possible include flatpak versions in your releases and to push your releases to flathub as well which is the main flatpak repo people turn to?

**Comment by danielweck** [comment]
https://www.electron.build/configuration/flatpak.html

> Currently electron-builder does not support publishing apps to Flatpak repositories like [Flathub](https://flathub.org/). This means the Flatpak support in electron-builder is limited to generating [single-file bundles](https://docs.flatpak.org/en/latest/single-file-bundles.html) which have various limitations compared to app bundles installed from a repository.

> For what it’s worth, there are [some](https://discourse.flathub.org/t/seeking-contractors-for-work-on-flathub-project/1889) [plans](https://discourse.flathub.org/t/is-it-possible-to-publish-a-self-contained-flatpak-file-to-flathub/2083) to make it easier to publish Electron apps to Flathub. When that happens, it should be easier to create a Flathub publisher for electron-builder (which would work similary to the other publishers).

**Comment by kevinekitabu** [comment]
@danielweck Really appreciate this. Has anyone gotten it to work? One of our partners requested this and we are having problems getting this to work.

**Comment by danielweck** [comment]
follow up https://github.com/edrlab/thorium-reader/discussions/2481

**Comment by mbridon** [comment]
Also, just a note, I tried the AppImage release, but couldn't figure out how to make it work and install it on Fedora 🤷‍♂️

A Flatpak app is much easier to work with, and with Flathub you get one click install straight from there 😍

**Comment by JimKillock** [comment]
Came here to make the same suggestion! And also to request apt packages. apt plus deb would cover most people.

**Comment by IcedQuinn** [comment]
Looks like this is only plausible by making the single-file bundle an artifact on Github or hosting one's own Flatpak repository. Flathub requires turning over the whole build process to them.

**Comment by mbridon** [comment]
Is it a problem that Flathub takes over the whole build process? 🤔 

It's how most apps are built for Flathub 😕

**Comment by danielweck** [comment]
https://www.electron.build/flatpak
